src/pylibzfs2.h

```diff
@@ -115,6 +115,16 @@ typedef struct {
 	PyObject *name;
 } py_zfs_pool_t;
 
+typedef struct {
+	PyObject_HEAD
+	py_zfs_t *pylibzfsp;
+	zpool_handle_t *zhp;
```

Comment:
I think rather than directly putting `zpool_handle_t` in here we should have a pointer to `py_zfs_pool_t` and do proper INCREF/DECREF of it. Otherwise we can't tell when it's safe to close the zpool handle.

Reply:
Updated to put in `py_zfs_pool_t` in place of `zpool_handle_t` and did `Py_INCREF` for it. Reference for `py_zfs_t` is also removed.